components/calendar/tests/monotonic_year.rs

```diff
@@ -0,0 +1,72 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_calendar::AnyCalendar;
+use icu_calendar::AnyCalendarKind;
+use icu_calendar::Date;
+
+#[cfg(test)]
+use std::rc::Rc;
+
+/// Reference: <https://tc39.es/proposal-intl-era-monthcode/#sec-temporal-calendardatearithmeticyear>
+static MONOTONIC_EPOCHS: &[(AnyCalendarKind, i32)] = &[
+    (AnyCalendarKind::Buddhist, -543),
+    (AnyCalendarKind::Chinese, 0),
+    (AnyCalendarKind::Coptic, 283),
+    (AnyCalendarKind::Dangi, 0),
+    (AnyCalendarKind::Ethiopian, 7),
+    (AnyCalendarKind::EthiopianAmeteAlem, -5493),
+    (AnyCalendarKind::Gregorian, 0),
+    (AnyCalendarKind::Hebrew, -3761),
+    (AnyCalendarKind::Indian, 78),
+    (AnyCalendarKind::HijriTabularTypeIIFriday, 621),
+    (AnyCalendarKind::HijriSimulatedMecca, 621),
+    (AnyCalendarKind::HijriTabularTypeIIThursday, 621),
+    (AnyCalendarKind::HijriUmmAlQura, 621),
+    (AnyCalendarKind::Iso, 0),
+    (AnyCalendarKind::Japanese, 0),
+    (AnyCalendarKind::JapaneseExtended, 0),
+    (AnyCalendarKind::Persian, 621),
+    (AnyCalendarKind::Roc, 1911),
+];
+
+#[test]
+fn test_monotonic_year() {
+    for (kind, monotonic_epoch) in MONOTONIC_EPOCHS.iter() {
+        let calendar = Rc::new(AnyCalendar::new(*kind));
+
+        let iso_date_in_epoch_year = Date::try_new_iso(*monotonic_epoch, 12, 31).unwrap();
```

Comment:
why is this 12-31? This might be the next calendar year for some lunar calendars

Reply:
Shane wrote this test. I think I would write it differently, I'll try.